test/assertions_test.rb

```diff
@@ -33,6 +33,24 @@ def test_assert_no_statsd_calls
     end
     assert_equal assertion.message, "No StatsD calls for metric counter expected."
 
+    @test_case.assert_no_statsd_calls('counter1', 'counter2') do
+      # noop
+    end
+
+    @test_case.assert_no_statsd_calls('counter1', 'counter2') do
+      StatsD.increment('counter')
+    end
+
+    assertion = assert_raises(Minitest::Assertion) do
+      @test_case.assert_no_statsd_calls('counter1', 'counter2') do
+        StatsD.increment('counter0')
+        StatsD.increment('counter1')
+        StatsD.increment('counter2')
+        StatsD.increment('counter3')
+      end
```

Comment:
Maybe a test where `assert_no_statsd_calls` has multiple metrics, but the block only emits one of them would be a good test to add that the logic is `any`, rather than `all`.